core/src/main/java/org/testcontainers/dockerclient/UnixSocketClientProviderStrategy.java

```diff
@@ -6,31 +6,24 @@
 import org.apache.commons.lang.SystemUtils;
 import org.jetbrains.annotations.NotNull;
 
-import java.io.File;
 import java.io.IOException;
 import java.nio.file.Files;
 import java.nio.file.Path;
 import java.nio.file.Paths;
 
 @Slf4j
 public class UnixSocketClientProviderStrategy extends DockerClientProviderStrategy {
```

Comment:
I ran some simple JMH benchmarks, comparing performance for both creating a container and just retrieving a container listing. **TLDR;** the docker daemon is the bottleneck for heavy operations, and there's a small performance boost with KQueue, but probably not significant especially given how infrequently we call it:
```
Benchmark                                             Mode  Cnt    Score     Error  Units
ClientPerfBenchmark.directClientCreatingContainers   thrpt   20    9.839 ±   0.340  ops/s
ClientPerfBenchmark.directClientJustListing          thrpt   20  435.923 ± 117.758  ops/s
ClientPerfBenchmark.proxiedClientCreatingContainers  thrpt   20   10.085 ±   0.663  ops/s
ClientPerfBenchmark.proxiedClientJustListing         thrpt   20  202.696 ±  43.040  ops/s
```

*However*, I'm coming around to using the KQueue route on macOS 10.12+, as it will help us establish confidence in this netty component for the future.